pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -630,6 +682,26 @@ func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, ru
 		}(container)
 	}
 	wg.Wait()
+
```

Comment:
One other option... we could also add a descriptively named unit test which will fail if this order is not respected. Such an addition would probably give us the strongest protections.

Reply:
I have struggled with coming up with thorough tests for this part as it is more dependent on ordering than the result of the function calls.  
Your suggestion would definitely make it easier to read, but I'm still not sure how to go about writing better tests for it. 
Also similar to my point about KillContainer, I'm cautious about interfering with existing flows, the current implementation does not effect the normal KillContainers flow as everything is feature gated. Perhaps I'm being overly cautious.

Reply:
Yeah, its definitely tricky. I'm curious to hear from others around the right level of caution... I think, given this PR is already a decent size, it may make sense to tackle this refactor later (whether during the alpha or during the beta/GA promotion)... Perhaps a comment capturing this conversation is the best short term solution?